llvm/lib/Target/EVM/EVMStackifyCodeEmitter.cpp

```diff
@@ -417,46 +407,42 @@ void EVMStackifyCodeEmitter::run() {
   SmallPtrSet<MachineBasicBlock *, 32> Visited;
   SmallVector<MachineBasicBlock *, 32> WorkList{&MF.front()};
   while (!WorkList.empty()) {
-    auto *Block = WorkList.pop_back_val();
-    if (!Visited.insert(Block).second)
+    auto *MBB = WorkList.pop_back_val();
+    if (!Visited.insert(MBB).second)
       continue;
 
     // Might set some slots to junk, if not required by the block.
-    CurrentStack = StackModel.getMBBEntryStack(Block);
-    Emitter.enterMBB(Block, CurrentStack.size());
+    CurrentStack = StackModel.getMBBEntryStack(MBB);
+    Emitter.enterMBB(MBB, CurrentStack.size());
+
+    SmallVector<const MachineInstr *> MIs;
+    for (const auto &MI : StackModel.instructionsToProcess(MBB))
+      MIs.push_back(&MI);
```

Comment:
Could you, please clarify why do we first need to put instructions into the vector `MIs` and then iterate over it, instead of just calling `emitMIEntryStack(MI)` directly?

Reply:
Added a comment before 3 lines above